### PR TITLE
DDF-5964 Upgraded to fmt-maven-plugin 2.9

### PR DESCRIPTION
**Pull request by shaundmorris**
#### What does this PR do?
Upgrades to fmt-maven-plugin 2.9
Google Java format upgrades from 1.5 to 1.7: https://github.com/google/google-java-format/releases
 - stream no longer is on its own line
 - fixed incorrect formatting for long annotations

#### Select relevant component teams: 
@codice/build 
@codice/continuous-integration  

#### Ask 2 committers to review/merge the PR and tag them here.
@mojogitoverhere 
@rzwiefel 

#### Any background context you want to provide?
Seems to run about 25% faster than current version of the plugin, not that the current one is slow. Talking about going from 20s to 15s :-)

#### What are the relevant tickets?
Fixes: #5964 

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

---
This change is [Reviewable](https://reviewable.io/reviews/codice/ddf/5965)

### CODE REVIEW

File: catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/CatalogCommands.java [hunk 1]
```diff
@@ -66,14 +66,13 @@
 
   // DDF-535: remove "-provider" alias in DDF 3.0
   @Option(
-    name = "--provider",
```
Comment (coyotesqrl) [inline catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/CatalogCommands.java:69]: ❓ Can the formatter be changed to only use a two-space indent for multi-line annotations? It'd be nice to reduce the files changed by what looks like an arbitrary style change.

File: catalog/core/catalog-core-camelcomponent/src/main/java/ddf/camel/component/catalog/content/ContentProducerDataAccessObject.java [hunk 2]
```diff
@@ -242,8 +242,7 @@ protected void logIds(List<Metacard> metacards, String action) {
       LOGGER.debug(
           "content item(s) {} with id = {}",
           action,
-          metacards
```
Comment (coyotesqrl) [inline catalog/core/catalog-core-camelcomponent/src/main/java/ddf/camel/component/catalog/content/ContentProducerDataAccessObject.java:245]: ❓ Can the formatter be changed to keep the call to `stream()` on its own line for multiline streaming blocks? It'd be nice to reduce the files changed by what looks like an arbitrary style change.

**Comment by shaundmorris** [comment]
> Is this using Google Java Format for its formatting rules? Did they change? Or were they improperly applied before? This just doesn't seem right.

@coyotesqrl yeah great questions. It made me start debating if an upgrade would be valuable. My understanding is they changed the formatting rules, but I haven't found great context as to why. We are currently using Google Java Format 1.5 and the latest is 1.7 (Jan 2019). You can see the latest release is where the stream change happened: https://github.com/google/google-java-format/releases.

**Comment by coyotesqrl** [comment]
> > Is this using Google Java Format for its formatting rules? Did they change? Or were they improperly applied before? This just doesn't seem right.
> 
> @coyotesqrl yeah great questions. It made me start debating if an upgrade would be valuable. My understanding is they changed the formatting rules, but I haven't found great context as to why. We are currently using Google Java Format 1.5 and the latest is 1.7 (Jan 2019). You can see the latest release is where the stream change happened: https://github.com/google/google-java-format/releases.

Ah. And I see that they changed the formatting on Annotations in 1.6 per [this issue](https://github.com/google/google-java-format/issues/159).

Okay. 👍

File: catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/CatalogCommands.java (hunk 1 above)
Comment (pklinef) [reply catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/CatalogCommands.java:69]: Won't changing the defaults make it drift from the various `fmt` IDE plugins?

**Comment by shaundmorris** [comment]
@pklinef 
> Won't changing the defaults make it drift from the various fmt IDE plugins?
Great point I believe we would also need to use the latest IDE plugins https://github.com/google/styleguide/blob/gh-pages/intellij-java-google-style.xml

I will verify this is works  together, update docs, and post some announcements on it.

File: catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/CatalogCommands.java (hunk 1 above)
Comment (shaundmorris) [reply catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/CatalogCommands.java:69]: Verified that the latest intellij plugin will format the same as the maven plugin. So need to be in sync on the versions. https://github.com/google/styleguide/blob/gh-pages/intellij-java-google-style.xml.

**Comment by shaundmorris** [comment]
build now

**Comment by shaundmorris** [comment]
I've held off on this one, since I would prefer to upgrade all fmt plugins across repos at the same time.

**Comment by LinkMJB** [comment]
> I've held off on this one, since I would prefer to upgrade all fmt plugins across repos at the same time.

@shaundmorris would that mean a change to a parent pom, instead?

**Comment by shaundmorris** [comment]
> @shaundmorris would that mean a change to a parent pom, instead?

@LinkMJB possibly. It seems like the development experience would be better to get on the same fmt plugin. However Alliance and DDF already have separate ones so maybe it isn't a big deal. I like putting it in the parent pom. Just would have to put up the reformat and the ddf-parent upgrade in the same PR to get them to work on all the repos.

**Comment by shaundmorris** [comment]
build now

**Comment by shaundmorris** [comment]
build now

**Pull request merged**